pkg/collector/corechecks/cluster/topologycollectors/container_correlator.go

```diff
@@ -63,11 +66,10 @@ func (*ContainerCorrelator) GetName() string {
 
 // Collects and Published the Cluster Component
 func (cc *ContainerCorrelator) CorrelateFunction() error {
-
-	nodeMap := make(map[string]string)
+	nodeMap := make(map[string]NodeIdentifierCorrelation)
 	// map containers that require the Node instanceId
 	for containerToNodeCorrelation := range cc.NodeIdentifierCorrChan {
-		nodeMap[containerToNodeCorrelation.NodeName] = containerToNodeCorrelation.NodeIdentifier
+		nodeMap[containerToNodeCorrelation.NodeName] = *containerToNodeCorrelation
```

Comment:
I'd suggest keep the map as `map[string]*NodeId...` to avoid copying here and after

Reply:
The problem is that later I need to use both `NodeIdentifier` and `NodeExternalID` from the nodeCorrelation.